docs/articles/documentation/guides/continuous-integration/github-actions-and-browserstack.md

```diff
@@ -0,0 +1,252 @@
+```
+
+{% endraw %}
+
+## Step 4 - Add a Step That Fetches The Repository
+
+Add a [step](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idsteps) that uses the [checkout](https://github.com/actions/checkout) action to fetch your repository content.
+
+{% raw %}
+
+```yml
+name: End-to-End Tests
+on: [push]
+env:
+        BROWSERSTACK_ACCESS_KEY: "${{ secrets.BROWSERSTACK_ACCESS_KEY }}"
+        BROWSERSTACK_USERNAME: "${{ secrets.BROWSERSTACK_USERNAME }}"
+jobs:
+  test:
+    name: Run TestCafe Tests
+    runs-on: windows-latest
+    steps:
+      - name: Check out the repository
+        uses: actions/checkout@v1
+```
+
+{% endraw %}
+
+## Step 5 - Add a Step To Install TestCafe BrowserStack plugin
+
+In order to enable TestCafe to run in the BrowserStack environment, install our [browserstack plugin](https://github.com/DevExpress/testcafe-browser-provider-browserstack).
+
+{% raw %}
+
+```yml
+name: End-to-End Tests
+on: [push]
+env:
+        BROWSERSTACK_ACCESS_KEY: "${{ secrets.BROWSERSTACK_ACCESS_KEY }}"
+        BROWSERSTACK_USERNAME: "${{ secrets.BROWSERSTACK_USERNAME }}"
+jobs:
+  test:
+    name: Run TestCafe Tests
+    runs-on: windows-latest
+    steps:
+      - name: Check out the repository
+        uses: actions/checkout@v1
+      - name: Install TestCafe BrowserStack plugin
+        run: npm install testcafe-browser-provider-browserstack
+```
+
+{% endraw %}
+
+## Step 6 - Add a Step That Runs TestCafe
+
+Add the [Run TestCafe](https://github.com/DevExpress/testcafe-action) action. Use the [args](#args) parameter to provide TestCafe [command line arguments](../../reference/command-line-interface.md), including the BrowserStack configuration:
+
+{% raw %}
+
+```yml
+name: End-to-End Tests
+on: [push]
+env:
+        BROWSERSTACK_ACCESS_KEY: "${{ secrets.BROWSERSTACK_ACCESS_KEY }}"
+        BROWSERSTACK_USERNAME: "${{ secrets.BROWSERSTACK_USERNAME }}"
+jobs:
+  test:
+    name: Run TestCafe Tests
+    runs-on: windows-latest
+    steps:
+      - name: Check out the repository
+        uses: actions/checkout@v1
+      - name: Run tests
+        uses: DevExpress/testcafe-action@latest
+        with:
+            args: "'browserstack:chrome@84.0:Windows 10' tests"
+```
+
+{% endraw %}
+
+> Note the additional pair of single quotes- BrowserStack configuration string includes a space, which can be interpreted as a separator by the command line.
+
+**Note**: Full list of available BrowserStack configurations can be obtained through console.
+First, you have to provide your credentials by exporting them:
+
+**bash**
+
+```sh
+export BROWSERSTACK_USERNAME="<your_browserstack_username>"
+export BROWSERSTACK_ACCESS_KEY="<your_browserstack_acess_key>"
+```
+
+**PowerShell**
+
+```sh
+Set-Variable -Name "BROWSERSTACK_USERNAME" -Value "<your_browserstack_username>"
+Set-Variable -Name "BROWSERSTACK_ACCESS_KEY" -Value "<your_browserstack_acess_key>"
+```
+
+Afterwards, run in console:
```

Comment:
What is the reader running in the console? TestCafe or BrowserStack, or something else?

Reply:
During the step above (exporting variables) built-in console commands are run, nothing to do with BrowserStack or TestCafe. That's why the commands differ for bash & powershell.

To pull the list of browsers (lines 180-184), user runs TestCafe

BrowserStack is not called directly anywhere during the course of this guide

Reply:
```suggestion
Then, run the following command in the console:
```